frontend/packages/dev-console/src/components/catalog/providers/useHelmCharts.tsx

```diff
@@ -0,0 +1,146 @@
+        const href = `/catalog/helm-install?chartName=${name}&chartRepoName=${chartRepositoryName}&chartURL=${encodedChartURL}&preselected-ns=${activeNamespace}`;
+
+        const maintainers = chart.maintainers?.length > 0 && (
+          <>
+            {chart.maintainers?.map((maintainer, index) => (
+              <React.Fragment key={index}>
+                {maintainer.name}
+                <br />
+                <a href={`mailto:${maintainer.email}`}>{maintainer.email}</a>
+                <br />
+              </React.Fragment>
+            ))}
+          </>
+        );
+
+        const homePage = chart.home && (
+          <ExternalLink href={chart.home} additionalClassName="co-break-all" text={chart.home} />
+        );
+
+        const detailsProperties = [
+          {
+            label: 'Chart Version',
+            value: version,
+          },
+          {
+            label: 'App Version',
+            value: appVersion,
+          },
+          {
+            label: 'Home Page',
+            value: homePage,
+          },
+          {
+            label: 'Maintainers',
+            value: maintainers,
+          },
+        ];
+
+        const helmChart = {
+          uid: digest,
+          type: 'HelmChart',
+          name: displayName,
+          description,
+          provider,
+          tags: keywords,
+          creationTimestamp: created,
+          attributes: {
+            name,
+            chartRepositoryName,
+            version,
+          },
+          icon: {
+            class: null,
+            url: imgUrl,
+          },
+          cta: {
+            label: 'Install Helm Chart',
+            href,
+          },
+          details: {
+            properties: detailsProperties,
+          },
+        };
+
+        // group Helm chart with same name and different version together
+        const existingChartIndex = normalizedCharts.findIndex((currentChart) => {
+          return (
+            currentChart.attributes?.name === name &&
+            currentChart.attributes?.chartRepositoryName === chartRepositoryName
+          );
+        });
+
+        if (existingChartIndex > -1) {
+          const existingChart = normalizedCharts[existingChartIndex];
+          const versionCompare = helmChart.attributes?.version?.localeCompare(
+            existingChart?.attributes?.version,
+            undefined,
+            { numeric: true, sensitivity: 'base' },
+          );
+
+          if (versionCompare === 1) {
+            normalizedCharts[existingChartIndex] = helmChart;
+          }
+        } else {
+          normalizedCharts.push(helmChart);
+        }
+      });
+      return normalizedCharts;
+    },
+    [] as CatalogItem[],
+  );
+};
+
+const useHelmCharts: CatalogExtensionHook<CatalogItem[]> = ({
+  namespace,
+}): [CatalogItem[], boolean, any] => {
+  const [helmCharts, setHelmCharts] = React.useState<HelmChartEntries>();
+  const [loaded, setLoaded] = React.useState<boolean>(false);
```

Comment:
Isn't `loaded === (helmCharts == null)` ?
Do we need a separate state?

Reply:
We can use helmCharts for loaded but i like to be explicit about these things.

Reply:
Done.